python/lsst/daf/butler/transfers/_yaml.py

```diff
@@ -247,6 +247,16 @@ def __init__(self, stream: IO, registry: Registry):
         self.calibAssociations: Dict[str, Dict[Timespan, List[DatasetId]]] = defaultdict(dict)
         self.refsByFileId: Dict[DatasetId, DatasetRef] = {}
         self.registry: Registry = registry
+
+        # If this is data exported before the reorganization of visits
+        # and visit systems and that new schema is in use, some filtering
+        # will be needed. The entry in the visit dimension record will be
+        # silently dropped when visit is created but the
+        # visit_system_membership must be constructed.
+        migrate_visit_system = False
+        if fileVersion < VersionTuple(1, 0, 2) and "visit_system_membership" in self.registry.dimensions:
```

Comment:
I'm not sure the file version is relevant here; consider someone creating an export file after this branch merges from a repo that hasn't been migrated.  I think it's the dimensions configuration version we need to start stuffing in the export files and using for logic like this.

Reply:
I'm not really sure what I was thinking. It's entirely true that <1.0.2 and modern registry does require a migration but you are right that we are still going to write the new version when exporting from the old registry.

Maybe the quick fix here is to make it write 1.0.1 as the version number if it's exporting an old registry?

The alternative is to store the dimensions version and namespace in the export YAML and read them on import. I'm starting to wonder now how we handle export/import for SPHEREx and PFS where they have a different namespace. Changing some dimension records is fine, but larger changes (as we did when adding visit_system_membership) are going to make this all blow up for other universes.

Reply:
Oh, and exporting from modern and trying to import into old is not going to work. Should it? Does that mean these export files should be version 2?

Reply:
I don't think there are any easy answers here - migration of YAML export files w.r.t. dimensions changes is just as hard in general as migration of databases, and we've been unclear on whether YAML export files are _more_ stable and migration-supported than databases (because many things that do affect databases don't affect them at all) or less (because they're _often_ more ephemeral and we don't have anything like alembic to lean on).

I do think we should start at least writing the dimensions namespace+version into export files, so we can tell when a migration may be needed.

And I think we should reserve the export file format version for changes to the import/export format itself - new kinds of fields it can have, changing some field from a `list` to `dict`, etc - and not use it as a stand-in for the dimensions config, _except_ as a way to guess the dimensions namespace+version when reading files that were written before we started saving the actual namespace+version.

Beyond that, maybe we need to start putting export YAML migration scripts for the transitions we care about somewhere, or just declaring that we don't support migrating those and they should really be transient things?

Reply:
Okay. I'll write the dimension version and namespace to the file. I will up the patch number for the file format but I won't use it -- I'll be able to use the version/patch for that test instead.

But yes, in some sense we should not only be writing `butler migrate` commands for migrating registries but also for migrating the export YAML files.

Reply:
Now looks like:
```
description: Butler Data Repository Export
version: 1.0.2
universe_version: 0
universe_namespace: daf_butler
```

I added a `DimensionUniverse.getVersionInfo()` method (but now I'm thinking I should make .version and .namespace read-only properties instead.

Reply:
Went with .version and .namespace.